sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkPatchItemRequestOptions.java

```diff
@@ -0,0 +1,88 @@
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.RequestOptions;
+import com.azure.cosmos.util.Beta;
+
+/**
+ * Encapsulates options that can be specified for an patch operation used in Bulk execution. It can be passed while
+ * creating bulk patch request using {@link BulkOperations}.
+ */
+@Beta(value = Beta.SinceVersion.V4_9_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class BulkPatchItemRequestOptions extends BulkItemRequestOptions{
+    private String filterPredicate;
+
+    /**
+     * Gets the FilterPredicate associated with the request in the Azure Cosmos DB service.
+     *
+     * @return the FilterPredicate associated with the request.
+     */
+    public String getFilterPredicate() {
+        return this.filterPredicate;
+    }
+
+    /**
+     * Sets the FilterPredicate associated with the request in the Azure Cosmos DB service.
+     *
+     * @param filterPredicate the filterPredicate associated with the request.
+     * @return the current request options
+     */
+    public BulkPatchItemRequestOptions setFilterPredicate(String filterPredicate) {
+        this.filterPredicate = filterPredicate;
+        return this;
+    }
+
+    /**
+     * Sets the boolean to only return the headers and status code in Cosmos DB response
+     * in case of Create, Update and Delete operations in {@link CosmosItemOperation}.
+     *
+     * If set to false, service doesn't return payload in the response. It reduces networking
+     * and CPU load by not sending the payload back over the network and serializing it on the client.
+     *
+     * This feature does not impact RU usage for read or write operations.
+     *
+     * By-default, this is null.
+     *
+     * NOTE: This flag is also present on {@link com.azure.cosmos.CosmosClientBuilder}, however if specified
+     * here, it will override the value specified in {@link com.azure.cosmos.CosmosClientBuilder} for this request.
+     *
+     * @param contentResponseOnWriteEnabled a boolean indicating whether payload will be included
+     * in the response or not for this operation.
+     *
+     * @return the current request options.
+     */
+    @Beta(value = Beta.SinceVersion.V4_9_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public BulkPatchItemRequestOptions setContentResponseOnWriteEnabled(Boolean contentResponseOnWriteEnabled) {
```

Comment:
@moderakh even derived overriding all composition seems not a great pattern. Any other alternatives. 
Its unrelated to this PR as such but as a concept/pattern.

Reply:
I know even I have done a lot of brainstorming on this thing. Please give any Ideas. I would be happy to learn.

Reply:
See my ocmment on the BulkItemRequestOptions change to remove final - I think the prposal there also addresses @kirankumarkolli  ' s feedback here.